jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/IdentityProviderAwareProcessListenerTest.java

```diff
@@ -102,14 +107,37 @@ public void testStartProcessWithIdentityListener() {
         final String userId = "userId";
         identityProvider.setName(userId);
 
-        long processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), "org.jbpm.writedocument");
+        long processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), PROCESS_ID);
         assertNotNull(processInstanceId);
         try {
-            final String initiator = (String) processService.getProcessInstanceVariable(processInstanceId, "initiator");
+            final String initiator = (String) processService.getProcessInstanceVariable(processInstanceId, PROCESS_INITIATOR_KEY);
             assertEquals(userId, initiator);
         } finally {
             processService.abortProcessInstance(processInstanceId);
         }
     }
 
+    @Test
+    public void testAbortProcessWithIdentityListener() {
+        assertNotNull(processService);
+
+        final String startUserId = "startUserId";
+        identityProvider.setName(startUserId);
+        long processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), PROCESS_ID);
+        ProcessInstance pi = processService.getProcessInstance(processInstanceId);
+
+        assertNotNull(processInstanceId);
+        final String initiator = (String) processService.getProcessInstanceVariable(processInstanceId, PROCESS_INITIATOR_KEY);
+        assertEquals(startUserId, initiator);
+
+        final String abortUserId = "abortUserId";
+        identityProvider.setName(abortUserId);
+        processService.abortProcessInstance(processInstanceId);
+
+        pi = processService.getProcessInstance(processInstanceId); // null
+        // final String terminator = (String) processService.getProcessInstanceVariable(processInstanceId, PROCESS_TERMINATOR_KEY); TODO find a way how to get variable from aborted process instance
+        //assertEquals(abortUserId, terminator);
+
```

Comment:
Collection<VariableDesc> variableHistory = runtimeDataService.getVariableHistory(processInstanceId, "terminator", null);
        assertThat(variableHistory).hasSize(1);
        VariableDesc var = variableHistory.iterator().next();
        assertThat(var.getVariableId()).isEqualTo("terminator");
        assertThat(var.getNewValue()).isEqualTo(abortUserId);
        
        List<InstanceView<?>> events = TestEventEmitter.getListEvents();
        List<ProcessInstanceView> views = events.stream().filter(ProcessInstanceView.class::isInstance).map(ProcessInstanceView.class::cast).collect(Collectors.toList());
        assertThat(views).hasSize(3);
        assertThat(views).extracting(ProcessInstanceView::getTerminator).anyMatch(e -> abortUserId.equals(e));

Reply:
you will need to change a bit the TestEventEmitter

Reply:
public class TestEventEmitter implements EventEmitter {


    private static List<InstanceView<?>> events = new ArrayList<>();
    
    @Override
    public void deliver(Collection<InstanceView<?>> data) {
        // no-op here as it is used to hook into active transaction
    }

    @Override
    public void apply(Collection<InstanceView<?>> data) {
        // store events generated
        events.addAll(data);
    }

    @Override
    public void drop(Collection<InstanceView<?>> data) {
        // no-op here are this impl reacts only on the apply method

    }

    @Override
    public EventCollection newCollection() {
        return new BaseEventCollection();
    }
    
    public static Set<InstanceView<?>> getEvents() {
        return new HashSet<>(events);
    }
    
    public static List<InstanceView<?>> getListEvents() {
        return events;
    }
    
    public static void clear() {
        events.clear();
    }

    @Override
    public void close() {
    }

}

Reply:
Done, thank you for your guidance!